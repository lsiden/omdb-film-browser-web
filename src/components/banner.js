import React from "react"
import QueryForm from "query-form"
import { OMDB_URL } from "constants"

const headerStyle = {
  background: "linear-gradient(to right, #000040, #3b00ff)",
  padding: "10px",
  color: "white",
}
const titleStyle = {
  fontSize: "18pt",
  marginBottom: "18pt",
}

const linkStyle = {
  color: "white",
  fontSize: "small",
}

const Banner = () => (
  <header style={headerStyle}>
    <h1 style={titleStyle}>
      {"Search Open Movie Database"}&nbsp;
      <a href={OMDB_URL} style={linkStyle}>
        (OMDB)
      </a>
    </h1>
    <QueryForm />
  </header>
)

export default Banner
