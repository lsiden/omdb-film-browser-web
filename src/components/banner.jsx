import React from 'react'

import { headerStyle, titleStyle } from 'style'
import { BANNER_TITLE } from 'omdb_constants'

const AboutLink = () => (
  <a href="/about" style={{ color: 'white' }}>
    about
  </a>
)

const inAboutPath = () => {
  return /about\/?$/.test(window.location.href)
}

export default () => {
  return (
    <header style={headerStyle}>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <h1 style={titleStyle}>{BANNER_TITLE}</h1>
        {!inAboutPath() && <AboutLink />}
      </div>
    </header>
  )
}