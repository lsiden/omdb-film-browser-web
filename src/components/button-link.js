import React from "react"
import PropTypes from "prop-types"

const ButtonLink = (props) => <button className="btn btn-link" {...props} />

ButtonLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string])
    .isRequired,
}

export default ButtonLink
