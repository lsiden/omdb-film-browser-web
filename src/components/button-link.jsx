import React from 'react'
import PropTypes from 'prop-types'

const ButtonLink = ({ style, onClick, children }) => (
  /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
  <button
    type="button"
    className="btn btn-link"
    style={style}
  >
    {children}
  </button>
)

ButtonLink.propTypes = {
  style: PropTypes.object,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
}

ButtonLink.defaultProps = {
  style: {},
}

export default ButtonLink
