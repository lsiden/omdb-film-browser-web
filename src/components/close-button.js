import React from "react"
import { Link } from "react-router-dom"

import ButtonLink from "components/button-link"

const CLOSE_ICON = "\u2715"

const closeButtonStyle = {
  margin: "8pt",
  fontSize: "20pt",
}

export const CloseButton = () => (
  <Link to="/">
    <ButtonLink style={closeButtonStyle} title="Close">
      {CLOSE_ICON}
    </ButtonLink>
  </Link>
)

export default CloseButton
