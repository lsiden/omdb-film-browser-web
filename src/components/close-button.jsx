import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import ButtonLink from 'components/button-link'

const CLOSE_ICON = '\u2715'

const closeButtonStyle = {
  margin: 20,
  fontSize: 20,
}

const CloseButton = ({ buttonStyle, onClick }) => {
  const style = {
    ...closeButtonStyle,
    ...buttonStyle,
  }
  return (
    <Link to="/">
      <ButtonLink onClick={onClick} style={style} title="Close">
        {CLOSE_ICON}
      </ButtonLink>
    </Link>
  )
}
CloseButton.propTypes = {
  buttonStyle: PropTypes.object,
}
CloseButton.defaultProps = {
  buttonStyle: {},
}

export default CloseButton
