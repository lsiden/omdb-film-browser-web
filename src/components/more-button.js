import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import Spinner from "respin"

import { pageFetch } from "actions/remote"

const spinnerStyle = {
  marginLeft: 45,
}
export const PureMoreButton = ({ isFetching, dispatchPageFetch }) =>
  isFetching ? (
    <div style={spinnerStyle}>
      <Spinner />
    </div>
  ) : (
    <button className="more-button" onClick={dispatchPageFetch}>
      More
    </button>
  )

PureMoreButton.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  dispatchPageFetch: PropTypes.func.isRequired,
}

export default connect(
  state => ({
    isFetching: !!state.isFetching,
  }),
  dispatch => ({
    dispatchPageFetch: () => dispatch(pageFetch()),
  })
)(PureMoreButton)
