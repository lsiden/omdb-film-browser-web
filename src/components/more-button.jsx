import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { promiseQueryPageFetch } from 'actions/remote'
import NavButton from 'components/nav-button'

const buttonStyle = {
  marginBottom: 20,
  fontSize: 20,
  color: 'darkgrey',
}

export const _MoreButton = ({ dispatchPageFetch }) => (
  <NavButton onClick={dispatchPageFetch} style={buttonStyle} title="More">
    More
  </NavButton>
)

_MoreButton.propTypes = {
  dispatchPageFetch: PropTypes.func.isRequired,
}

export default connect(
  () => ({}),
  (dispatch) => ({
    dispatchPageFetch: () => dispatch(promiseQueryPageFetch()),
  }),
)(_MoreButton)
