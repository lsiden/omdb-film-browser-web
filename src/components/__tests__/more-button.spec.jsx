import React from 'react'
import { create, act } from 'react-test-renderer';

import { _MoreButton } from 'components/more-button'

const createWrapper = (films=[], totalResults=0, dispatchPageFetch=() => {}) => create(
  <_MoreButton
    dispatchPageFetch={dispatchPageFetch}
    films={films}
    totalResults={totalResults}
  />
).root

describe('MoreButton', () => {
  it('renders null if no films', () => {
    const wrapper = createWrapper()
    expect(wrapper.findAllByType('button').length).toBe(0)
  })

  it('renders null if films.length <= totalResults', () => {
    const films = Array(3)
    const wrapper = createWrapper(films, films.length)
    expect(wrapper.findAllByType('button').length).toBe(0)
  })

  it('matches snapshot if films.length < totalResults', () => {
    const films = Array(3)
    expect(createWrapper(films, films.length+1)).toMatchSnapshot()
  })

  it('calls dispatchPageFetch', () => {
    const dispatchPageFetch = jest.fn()
    const films = Array(3)
    const wrapper = createWrapper(films, films.length+1, dispatchPageFetch)
    const btn = wrapper.findByType('button')
    act(btn.props.onClick)
    expect(dispatchPageFetch.mock.calls.length).toBe(1)
  })
})
