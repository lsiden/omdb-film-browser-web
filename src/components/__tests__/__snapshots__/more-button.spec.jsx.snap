// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreButton matches snapshot 1`] = `
<button
  className="btn btn-link"
  onClick={[MockFunction]}
  style={
    Object {
      "color": "darkgrey",
      "fontSize": 20,
      "marginBottom": 20,
    }
  }
  type="button"
>
  More
</button>
`;
